runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaCacheClientFetchFactory.java

```diff
@@ -1164,6 +1178,7 @@ private void doClientReplyData(
         {
             assert nextEntry != null;
 
+            group.dataFrameReceived = true;
```

Comment:
Consider the case with 2 client streams, each one would need to receive a `FLUSH` if there are no `DATA` frames sent before catching up to latest / stable.

Therefore, we cannot track this at the group level as the first client would get the right behavior, but all future clients would not get the `FLUSH` as they would count the first client's `DATA` frame as their own.